error.go

```diff
@@ -0,0 +1,69 @@
+package semver
+
+import (
+	"bytes"
+	"fmt"
+)
+
+var rangeErrs = [...]string{
+	"%s is less than the minimum of %s",
+	"%s is less than or equal to the minimum of %s",
+	"%s is greater than the maximum of %s",
+	"%s is greater than or equal to the maximum of %s",
+	"%s is specifically disallowed in %s",
+}
+
+const (
+	rerrLT = iota
+	rerrLTE
+	rerrGT
+	rerrGTE
+	rerrNE
+)
+
+type MatchFailure interface {
+	error
+	Pair() (v *Version, c Constraint)
+}
```

Comment:
Originally, it was. I've debated this (with myself) over in `vsolver`, as well. In both cases, though, I think there's a distinction between a *failure* and an *error*. It's more visible there - for example, if there's a permissions issue with writing the cache to disk, that's an *error*. But if you can't reach a solution due to incompatible constraints, it's a *failure*.

Because this is a closed, purely numerical and complete (as much as Gödel would allow) system,   think, there's no real opportunity for *errors*. Just routine failures.

Like I said, I've been arguing with myself about this. If you'd prefer it be Error, that's fine, we can change it back.